packages/utilities/src/classNamesFunction.ts

```diff
@@ -85,7 +91,9 @@ export function classNamesFunction<TStyleProps extends {}, TStyleSet extends ISt
       typeof styleFunctionOrObject === 'function' &&
       (styleFunctionOrObject as StyleFunction<TStyleProps, TStyleSet>).__noStyleOverride__
     ) {
-      return styleFunctionOrObject(styleProps) as IProcessedStyleSet<TStyleSet>;
+      return (styleFunctionOrObject as IStyleFunction<TStyleProps, TStyleSet>)(
```

Comment:
had to add additional casting, failing after upgrade with this TS2349: 

![Screenshot 2024-12-20 at 17 08 41](https://github.com/user-attachments/assets/4c7abf40-4659-4108-99f9-b2a3a8d78840)